### DISCUSSION TITLE
How can I validate and serialize object myslef in the same way Fastify does when handling route?

### DISCUSSION THREAD

**Discussion by vatosarmat** [fastify/fastify#5473] [Q&A] [answered]
Hello. My application also sends some data by websocket and I would like it to be validated and serialized in the same way with the same schema I set it up in my route. What is the easiest and most idiomatic way? Thanks.

**Comment by metcoder95** [comment]
Just to understand better, you want to validate the incoming `ws` payloads after the upgrade has been done?

**Reply by vatosarmat** [reply]
I don't interact with `ws` upgrades and other low-level details, I'm just sending data to `socket.io` socket and would like to reuse validation logic from normal HTTP REST routes for that data.
It seems that I have to create my own `Ajv` instance and compile schemas by myself, but I thought that maybe it is possible to extract them from `FastifyInstance`? 
Just found that `FastifyInstance` has `validatorCompiler` and `serializerCompiler` and seems like I should compile my schemas with them, but they both `undefined`

**Comment by vatosarmat** [comment, marked as answer]
Here it is https://github.com/fastify/fastify/issues/3897
`Request` and `Reply` have appropriate methods
